nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java

```diff
@@ -155,10 +153,32 @@ public ClassLoader getCommonClassLoader() {
         return commonClassLoader != null ? commonClassLoader : APIClassLoader;
     }
 
+    /**
+     * Adds a classpath element to the common classloader if the classloader supports it.
+     * @param url URL of the classpath element to add
+     * @throws UnsupportedOperationException If adding not supported by the classloader
+     */
+    public void addToClassPath(URL url) {
+        if (commonClassLoader != null) {
+            commonClassLoader.addURL(url);
+        } else {
+            commonClassLoader = new GlassfishUrlClassLoader(new URL[] {url}, APIClassLoader);
+        }
+        commonClassPath = urlsToClassPath(Arrays.stream(commonClassLoader.getURLs()));
+    }
+
     public String getCommonClassPath() {
         return commonClassPath;
     }
 
+    private static String urlsToClassPath(Stream<URL> urls) {
+        return urls
+                .map(URL::getFile)
+                .filter(Predicate.not(String::isBlank))
+                .map(file -> new File(file).getAbsolutePath())
```

Comment:
Could the lambda be replaced with additional `.map`?

Reply:
You mean have 2 map calls instead of just one? It could but I thought it’s an unnecessary overhead to call map twice.

Reply:
> You mean have 2 map calls instead of just one? It could but I thought it’s an unnecessary overhead to call map twice.

Yes, yes.

2 map calls instead of one, and no lambda instead of one.